### PR TITLE
feat(delete operation): warning before deletion

### PR DESCRIPTION
**Pull request by CatalinB7**

### CODE REVIEW

File: budget-manager/src/app/category-modal/category-modal.component.ts [hunk 1]
```diff
@@ -99,11 +104,25 @@ export class CategoryModalComponent {
...
   openSnackBar(message: string, duration: number) {
-    this._snackBar.open(message,'', { duration });
+    this._snackBar.open(message, '', { duration });
+  }
+
+
+  openDialogDelWarn(toDelete: string) {
+    const dialogRef = this._dialog.open(DeleteWarningDialogComponent, {
+      width: '35vw',
+      height: '20vh',
```
Comment (dochiasebastian) [inline budget-manager/src/app/category-modal/category-modal.component.ts:118]: Isn't it a bit large for a simple message?

File: budget-manager/src/app/delete-warning-dialog/delete-warning-dialog.component.scss [hunk 2]
```diff
@@ -0,0 +1,12 @@
+@import "../../variables.scss";
+
+.distanced-btns {
+    width: 100%;
+    display: flex;
+    justify-content: space-evenly;
+    align-items: center;
+}
```
Comment (dochiasebastian) [inline budget-manager/src/app/delete-warning-dialog/delete-warning-dialog.component.scss:8]: image
It looks pretty bad.

Comment (dochiasebastian) [reply budget-manager/src/app/delete-warning-dialog/delete-warning-dialog.component.scss:8]: and the continue button can be 'warn'

**Review by dochiasebastian — APPROVED** [review]

File: budget-manager/src/app/delete-warning-dialog/delete-warning-dialog.component.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+} from '@angular/core';
+import {
+  MAT_DIALOG_DATA,
+  MatDialogRef,
+} from '@angular/material/dialog';
+
+@Component({
+  selector: 'app-delete-warning-dialog',
+  templateUrl: './delete-warning-dialog.component.html',
+  styleUrls: ['./delete-warning-dialog.component.scss']
```
Comment (anbalase) [inline budget-manager/src/app/delete-warning-dialog/delete-warning-dialog.component.ts:13]: It's a new component, why oh why don't you use onPush? 😄

**Pull request merged**